### PR TITLE
Update pre-commit and lint

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR updates pre-commit configuration and runs it to lint the code and clean up any styling issues.

### CODE REVIEW

File: ooipy/request/authentification.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """
-This module provides functions for the automatic authentification at the
-OOI. The authentification is based on a username and a token, which will
+This module provides functions for the automatic authentication at the
+OOI. The authentication is based on a username and a token, which will
```
Comment (lsetiawan) [inline ooipy/request/authentification.py:3]:
```suggestion
This module provides functions for the automatic authentification at the
OOI. The authentification is based on a username and a token, which will
```

File: ooipy/request/ctd_request.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 import requests
 
-import ooipy.request.authentification
+import ooipy.request.authentication
```
Comment (lsetiawan) [inline ooipy/request/ctd_request.py:7]:
```suggestion
import ooipy.request.authentification
```

File: ooipy/request/ctd_request.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def get_ctd_data(
 
     """
 
-    USERNAME, TOKEN = ooipy.request.authentification.get_authentification()
+    USERNAME, TOKEN = ooipy.request.authentication.get_authentification()
```
Comment (lsetiawan) [inline ooipy/request/ctd_request.py:74]:
```suggestion
    USERNAME, TOKEN = ooipy.request.authentification.get_authentification()
```

File: tests/test_request.py [hunk 4]
```diff
@@ -95,7 +95,7 @@ def test_hydrophone_node_names():
         assert hyd_data.stats.location in node_id_arr
 
 
-# TODO: need to figure out how to do the OOI authentification on the GitHub VM
+# TODO: need to figure out how to do the OOI authentication on the GitHub VM
```
Comment (lsetiawan) [inline tests/test_request.py:98]:
```suggestion
# TODO: need to figure out how to do the OOI authentification on the GitHub VM
```

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -10,27 +15,25 @@ repos:
...
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v2.38.0
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.2
     hooks:
-    - id: pyupgrade
-      args: ["--py36-plus"]
+      - id: codespell
+        # Ignores `.ipynb` files, `_build` and `_static` folders
+        args: ["--skip=*.ipynb,docs/source/_build,docs/source/_static", "-w", "docs", "ssec_project"]
```
Comment (lsetiawan) [inline .pre-commit-config.yaml:39]:
```suggestion
        args: ["--skip=*.ipynb,docs/source/_build,docs/source/_static", "--ignore-words=authentification", "-w", "docs", "ooipy"]
```

File: ooipy/__init__.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 import ooipy.tools.workflow
 from ooipy.hydrophone.basic import Psd, Spectrogram
-from ooipy.request.authentification import set_authentification
+from ooipy.request.authentication import set_authentification
```
Comment (lsetiawan) [inline ooipy/__init__.py:5]:
```suggestion
from ooipy.request.authentification import set_authentification
```

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -10,27 +15,25 @@ repos:
...
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v2.38.0
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.2
     hooks:
-    - id: pyupgrade
-      args: ["--py36-plus"]
+      - id: codespell
+        # Ignores `.ipynb` files, `_build` and `_static` folders
+        args: ["--skip=*.ipynb,docs/source/_build,docs/source/_static", "--ignore-words=authentification", "-w", "docs", "ooipy"]
```
Comment (lsetiawan) [inline .pre-commit-config.yaml:39]:
```suggestion
        args: ["--skip=*.ipynb,docs/source/_build,docs/source/_static", "--ignore-words-list=authentification", "-w", "docs", "ooipy"]
```

**Review by John-Ragland — APPROVED** [review]
All looks good, thanks for the updates!

**Pull request merged**